types/prettier/index.d.ts

```diff
@@ -22,16 +22,16 @@ export type LiteralUnion<T extends U, U = string> = T | (Pick<U, never> & { _?:
 export type AST = any;
 export type Doc = doc.builders.Doc;
 
-// https://github.com/prettier/prettier/blob/master/src/common/fast-path.js
-export interface FastPath<T = any> {
+// https://github.com/prettier/prettier/blob/main/src/common/ast-path.js
+export interface AstPath<T = any> {
     stack: T[];
     getName(): null | PropertyKey;
     getValue(): T;
     getNode(count?: number): null | T;
     getParentNode(count?: number): null | T;
     call<U>(callback: (path: this) => U, ...names: PropertyKey[]): U;
-    each(callback: (path: this) => void, ...names: PropertyKey[]): void;
-    map<U>(callback: (path: this, index: number) => U, ...names: PropertyKey[]): U[];
+    each(callback: (path: this, index: number, value: any) => void, ...names: PropertyKey[]): void;
+    map<U>(callback: (path: this, index: number, value: any) => U, ...names: PropertyKey[]): U[];
```

Comment:
I need to find out the types of `value`

Reply:
Are these `value`s `T`?

Reply:
I think that's true in most cases, but it could be array of anything in theory.  For example:

`new AstPath({foo:[1,true,null]}).map((_,__, value) => console.log(value))`, `value` is `[1,true,null]` in this case.

Reply:
Maybe it's okay to just don't type it for now, but if someone uses these, they can improve the types on demand